src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -99,7 +102,8 @@ internal void RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment re
             batchBuilder,
             ComponentId,
             _nextRenderTree.GetFrames(),
-            CurrentRenderTree.GetFrames());
+            CurrentRenderTree.GetFrames(),
+            CurrentRenderTree.GetNamedEvents());
```

Comment:
GetNamedEvents() always returns null unless the renderer is tracking a named event handler.